### PR TITLE
boost: Updates package to version 1.81.0

### PR DESCRIPTION
**Pull request by ClaymorePT**
Maintainer:  @ClaymorePT 
Compile tested: Octeon PLus
Run tested: N/A

Description:

This commit updates boost to version 1.81.0

A new library is available:
- URL [[1]]: A library for parsing, modifying, and printing URLs using only C++11, from Vinnie Falco and Alan de Freitas. Features include fast compilation, strong invariants, and strict compliance using a memory-friendly approach.

More info about Boost 1.81.0 can be found at the usual place [[2]].

[1]: https://www.boost.org/doc/libs/1_81_0/libs/url/doc/html/index.html
[2]: https://www.boost.org/users/history/version_1_81_0.html

Signed-off-by: Carlos Miguel Ferreira <carlosmf.pt@gmail.com>

### CODE REVIEW

**Comment by ClaymorePT** [comment]
@neheb Some help here. These build failures are abnormal to me. Anything I can resolve on the boost side ?

**Comment by neheb** [comment]
Nah it’s fine.

File: libs/boost/Makefile [hunk 1]
```diff
@@ -11,13 +11,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=boost
-PKG_VERSION:=1.80.0
-PKG_SOURCE_VERSION:=1_80_0
-PKG_RELEASE:=$(AUTORELEASE)
+PKG_VERSION:=1.81.0
+PKG_SOURCE_VERSION:=1_81_0
+PKG_RELEASE:=4
```
Comment (1715173329) [inline libs/boost/Makefile:16]: It should be reset to `1` for version bumping.

Comment (ClaymorePT) [reply libs/boost/Makefile:16]: I was following what the bot said to do.

Comment (ClaymorePT) [reply libs/boost/Makefile]: Updated in 6c35386

Comment (1715173329) [reply libs/boost/Makefile]: Yup, sadly the bot doesn't seem to work properly ;(

**Comment by ClaymorePT** [comment]
Up to you now @neheb :)

**Pull request merged**